share/commands/agent-run

```diff
@@ -9,31 +9,103 @@
 # @man +
 # @man *-v*: run the agent in verbose mode, prints detailed information
 # @man +
+# @man *-d*: run the agent in debug mode, prints very detailed information
+# @man +
 # @man *-c*: run the agent without color output
+# @mab +
+# @man *-r*: run the agent with raw output
+# @mab +
+# @man *-m*: run the agent with multiline output
 # @man +
 # @man *-b*: run the agent on a specific bundle
+# @man +
+# @man *-D*: define a class for this run
 
 VERBOSITY=""
-COLOR="-C"
+COLOR="-Calways"
 BUNDLE=""
+CLASS=""
+MULTILINE=0
+SUMMARY_ONLY=0
+DEBUG_CLASS="-D debug"
+DISPLAY_INFO=0
+PRETTY_FILTER="/opt/rudder/share/lib/reports.awk"
+
+UUID=$(cat /opt/rudder/etc/uuid.hive 2>/dev/null)
+[ $? -ne 0 ] && UUID="Not yet configured"
+
+## Colors configuration (enable colors only if stdout is a terminal)
+if [ -t 1 ]; then
+    GREEN="\\033[1;32m"
+    DARKGREEN="\\033[0;32m"
+    RED="\\033[1;31m"
+    BLUE="\\033[1;34m"
+    TPINK="\\033[1;35m"
+    WHITE="\\033[0;02m"
+    WHITELIGHT="\\033[1;08m"
+    YELLOW="\\033[1;33m"
+    CYAN="\\033[1;36m"
+    NORMAL="\\033[0;39m"
+else
+    GREEN=""
+    DARKGREEN=""
+    RED=""
+    BLUE=""
+    PINK=""
+    WHITE=""
+    WHITELIGHT=""
+    YELLOW=""
+    CYAN=""
+    NORMAL=""
+fi
+
+VERSION=`rudder agent version`
 
-while getopts "iIvdcb:" opt; do
+PRETTY="awk -v uuid=\"\${UUID}\" -v info=\"\${DISPLAY_INFO}\" -v summary_only=\"\${SUMMARY_ONLY}\" -v multiline=\"\${MULTILINE}\" -v version=\"\${VERSION}\" \
+            -v green=\"\${GREEN}\" -v darkgreen=\"\${DARKGREEN}\" -v red=\"\${RED}\" -v yellow=\"\${YELLOW}\" -v normal=\"\${NORMAL}\" -v white=\"\${WHITE}\" -v cyan=\"\${CYAN}\" \
+            -f ${PRETTY_FILTER}"
+
+while getopts "iIvdrmcb:D:" opt; do
   case $opt in
     i|I)
-      VERBOSITY="-I -D debug"
+      VERBOSITY="-I ${DEBUG_CLASS}"
+      DISPLAY_INFO=1
+      MULTILINE=1
```

Comment:
Displays information messages from Rudder output inside the logs, so we force multiline.